finch/processes/wps_sdba.py

```diff
@@ -0,0 +1,220 @@
+        "is created when grouping data."
+        "Units are the sampling frequency of the data along the main dimension.",
+        data_type="integer",
+        default=1,
+        min_occurs=0,
+    ),
+    interp=LiteralInput(
+        "interp",
+        "Quantile interpolation",
+        abstract="Interpolation method to determine the quantile correction factor to apply. ",
+        data_type="string",
+        default="nearest",
+        allowed_values=["nearest", "linear", "cubic"],
+        min_occurs=0,
+    ),
+)
+adjust_args = dict(
+    extrapolation=LiteralInput(
+        "extrapolation",
+        "Extrapolation method",
+        abstract="Method used to extrapolate quantile adjustment factors beyond the computed quantiles.",
+        data_type="string",
+        default="constant",
+        allowed_values=["constant", "nan"],
+        min_occurs=0,
+    )
+)
+variable = wpsio.variable
+
+resources = dict(
+    ref=ComplexInput(
+        "ref",
+        "Reference dataset",
+        abstract="Reference netCDF resource, typically observations or reanalyses.",
+        max_occurs=1,
+        supported_formats=[FORMATS.NETCDF, FORMATS.DODS],
+    ),
+    hist=ComplexInput(
+        "hist",
+        "Historical simulation",
+        abstract="Historical simulation netCDF resource overlapping the reference dataset.",
+        max_occurs=1,
+        supported_formats=[FORMATS.NETCDF, FORMATS.DODS],
+    ),
+    sim=ComplexInput(
+        "sim",
+        "Target simulation to be corrected",
+        abstract="Simulation netCDF resource whose bias will be corrected",
+        max_occurs=1,
+        supported_formats=[FORMATS.NETCDF, FORMATS.DODS],
+    ),
+)
+
+common_outputs = [
+    ComplexOutput(
+        "output",
+        "netCDF output",
+        as_reference=True,
+        supported_formats=[FORMATS.NETCDF],
+    ),
+    wpsio.output_log,
+    wpsio.output_metalink,
+]
+
+
+class EmpiricalQuantileMappingProcess(FinchProcess):
+    def __init__(self):
+        inputs = (
+            list(resources.values())
+            + list(group_args.values())
+            + list(adjust_args.values())
+            + [
+                wpsio.variable_any,
+                LiteralInput(
+                    "nquantiles",
+                    "Number of quantile bins",
+                    abstract="The number of quantiles to use. Two endpoints at 1e-6 and 1 - 1e-6 will be added.",
+                    data_type="integer",
+                    default=20,
+                    min_occurs=0,
+                ),
+                LiteralInput(
+                    "kind",
+                    "Kind of adjustment (+, *)",
+                    abstract="Use * for multiplicative adjustment, or + for additive adjustement.",
+                    data_type="string",
+                    default="+",
+                    allowed_values=["+", "*"],
+                    min_occurs=0,
+                ),
```

Comment:
If this is too much trouble, it could also be a boolean? I don't think we'll implement other "kinds", as the _new way_ will arrive first.